# Урок 6. Сравнение долей. Построение доверительного интервала

In [1]:
import numpy as np
import scipy.stats as stats

***
## Задание 1.

### Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания $a$ с надежностью 0.95, если выборочная средняя $M = 80$, а объем выборки $n = 256$.

Поскольку нам известно среднее квадратическое отклонение генеральной совокупности $\sigma = 16$, то для расчета 95%-го доверительного интервала, будем использовать Z-критерий. Границы доверительного интервала расчитаем по формуле: \
$\bar{x} \pm z_{\frac{\alpha}{2}} \cdot \cfrac{\sigma}{\sqrt{n}}$

In [2]:
sd, Mx, N, gamma = 16, 80, 256, 0.95
alpha = 1 - gamma

z = stats.norm.ppf(1-alpha/2)
z

1.959963984540054

In [3]:
se = sd / np.sqrt(N)
se

1.0

In [4]:
print(f'{Mx-z*se:.2f} < a < {Mx+z*se:.2f}')

78.04 < a < 81.96


**Ответ**: С надежностью $0.95$ можно утверждать, что доверительный интервал $(78.04, 81.96)$ покрывает математическое ожидание $a$ генеральной совокупности.

***
## Задание 2.

### В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1. Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Поскольку из условия задачи нам ничего не известно ни о мат. ожидании генеральной совокупности, ни о среднем квадратическом отклонении для генеральной совокупности, то для расчета 95%-го доверительного интервала будем использовать t-критерий, и формулу: \
$\bar{x} \pm t_{\frac{\alpha}{2}} \cdot \cfrac{S}{\sqrt{n}}$

In [5]:
X = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])
n, gamma = 10, 0.95

Mx = np.mean(X)
Mx

6.590000000000001

In [6]:
D = np.var(X, ddof=1)
D

0.2032222222222223

In [7]:
se = np.sqrt(D/n)
se

0.14255603186895402

In [8]:
alpha = 1 - gamma
#t = abs(stats.t.ppf((1-gamma)/2, n-1))
t = stats.t.ppf(1 - alpha/2, n-1)
t

2.2621571627409915

In [9]:
print(f'{Mx-t*se:.2f} < X < {Mx+t*se:.2f}')

6.27 < X < 6.91


**Ответ**: С надежностью $0.95$ можно утверждать, что доверительный интервал $(6.27, 6.91)$ покрывает истинное значение величины $X$.

***
## Задание 3.

### Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170. Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175. Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

$\Delta \pm t_{\frac{\alpha}{2}} \cdot \cfrac{S_{\Delta}}{\sqrt{n}}$, где $\Delta = \bar{X_1} - \bar{X_2}$

$D = \cfrac{D_1+D_2}{2}$

$S_{\Delta} = \sqrt{\cfrac{D}{n_1}+\cfrac{D}{n_2}}$

$df = n_1 + n_2 - 2$

In [ ]:
X = np.array([178, 165, 165, 173, 168, 155, 160, 164, 178, 175])
Y = np.array([175, 167, 154, 174, 178, 148, 160, 167, 169, 170])
gamma = 0.95

In [12]:
n1, n2 = X.shape[0], Y.shape[0]
d = np.mean(X) - np.mean(Y)
d

1.9000000000000057

In [13]:
D = (np.var(X, ddof=1) + np.var(Y, ddof=1)) / 2
D

75.58333333333334

In [14]:
Sd = np.sqrt(D/n1 + D/n2)
Sd

3.8880157750022915

In [15]:
alpha = 1 - gamma
t = stats.t.ppf(1 - alpha/2, n1+n2-2)
t

2.10092204024096

In [16]:
print(f'{d-t*Sd:.2f} < M(X)-M(Y) < {d+t*Sd:.2f}')

-6.27 < M(X)-M(Y) < 10.07


**Ответ**: С надежностью $0.95$ можно утверждать, что доверительный интервал $(-6.27, 10.07)$ покрывает математическое ожидание разности среднего роста матерей и детей.